test/file-upload-quirks.test.ts

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import type { VersionInfo } from '@jsdevtools/host-environment';
+import type { Express } from 'express';
+
+import 'isomorphic-fetch';
+import { host } from '@jsdevtools/host-environment';
+
+import express from 'express';
+import multer from 'multer';
+import tempDirectory from 'temp-dir';
+import DatauriParser from 'datauri/parser';
+import { promises as fs } from 'fs';
+import { expect } from 'chai';
+import { FormDataEncoder } from 'form-data-encoder';
+
+import owlbertDataURL from './fixtures/owlbert.dataurl.json';
+import owlbertShrubDataURL from './fixtures/owlbert-shrub.dataurl.json';
+
+import arrayOfOwlbertsHAR from './fixtures/array-of-owlberts.har.json';
+
+const isNode18 = (host.node as VersionInfo).version >= 18;
+
+describe('#fetchHAR (Node-only quirks)', function () {
+  let fetchHAR;
+  let app: Express;
+  let listener;
+
+  beforeEach(async function () {
+    /**
+     * Under Node 18's native `fetch` implementation if a `File` global doesn't exist it'll polyfill
+     * its own implementation. Normally this works fine, but its implementation is **different**
+     * than the one that `formdata-node` ships and when we use the `formdata-node` one under Node
+     * 18 `type` options that we set into `File` instances don't get picked up, resulting in
+     * multipart payloads being sent as `application/octet-stream` instead of whatever content type
+     * was attached to that file.
+     *
+     * This behavior also extends to Undici's usage of `Blob` as well where the `Blob` that ships
+     * with `formdata-node` behaves differently than the `Blob` that is part of the Node `buffer`
+     * module, which Undici wants you to use.
+     */
+    if (isNode18) {
```

Comment:
I feel like I've seen similar code from you before - is this worth breaking into a util?

Reply:
Yeah this same code is duplicated in a few other tests here. I might refactor it.